pulsar-transaction/coordinator/src/main/java/org/apache/pulsar/transaction/coordinator/impl/MLTransactionMetadataStore.java

```diff
@@ -257,7 +257,7 @@ public void handleMetadataEntry(Position position, TransactionMetadataEntry tran
                         State.Ready, getState(), "add produced partition"));
                 return;
             }
-            getTxnPositionPair(txnID).thenAccept(txnMetaListPair -> {
+            getTxnPositionPair(txnID).thenCompose(txnMetaListPair -> {
```

Comment:
The origin method `getTxnPositionPair(txnID)` may throw TransactionNotFoundException, but it does not catch it. It may cause the uncompleted future. LGTM

Reply:
Yes, thanks for your better explaination.